proposals/large-ipld-dags.md

```diff
@@ -0,0 +1,117 @@
+# Support Large IPLD/IPFS DAGs
+
+Authors: Stebalien
+
+Initial PR: TBD <!-- Reference the PR first proposing this document. Oooh, self-reference! -->
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+_Describe the desired state of the world after this project? Why does that matter?_
+
+First, it should be possible to store arbitrary and arbitrarily large IPLD DAGs on Filecoin using
+the built-in protocols. At the moment, Filecoin can only store "whole DAGs". If a DAG, doesn't fit
+into a sector when serialized as a CAR, it must be converted to raw-blocks, chunked, and then stored
+as those chunks.
+
+Unfortunately:
+
+1. This workaround erases the underlying DAG structure. This makes it difficult to transfer this
+   data for both storage and retrieval. This is especially true when interacting with IPFS.
+2. This workaround requires storing an "overlay" DAG in Filecoin (paying for that storage).
+
+Second, it should be possible to retrieve subsets of DAGs. While the underlying protocols support
```

Comment:
The protocols support this - I think this is referring to graphsync and the other IPLD pieces down to the data storage - but the CLI doesn't. What about the miner side of this? The wording of this suggests that it's just the client CLI that's blocked on this, is that true? Can an alternative retrieval client use the protocols today to retrieve an arbitrary sub-DAG from a miner or is there more to be done on that side too?

Reply:
I tested selector-based retrievals way back in August ( using a hardcoded selector in the client directly ) - they worked, in the context of everything else being flaky.

It's not a CLI issue, rather we do not have a decent selector interchange format in general ( a gob of cbor is not something to use over API/CLI )

In other words:
- if today I want to specify a cid - I usually get to do the funny `{ "/":"baf..." }` thing
- if today I want to express a selector - I do... ❓

Reply:
ahhhhh back to the "selector syntax" problem, we should just solve that properly eh? so close https://github.com/ipld/specs/pull/239